DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -60,173 +63,208 @@ void Usage(string programName) {
+                    size_t num_point, float C) {
+  string path = prefix + "_error_diff.txt";
+  float delta = 0.001f;
+  float difference = 0.00f;
+  double norm2 = 0;
+  bool err = false;
 
-  try {
-    // Define the device queue
-    queue q = default_selector{};
-    cout << "Kernel runs on " << q.get_device().get_info<info::device::name>()
-         << "\n";
-
-    // Set boundary condition at one end.
-    arr[0] = arr_next[0] = temp;
-
-    float* current_data_ptr = arr;
-    float* next_data_ptr = arr_next;
-
-    // Buffer scope
-    {
-      buffer temperature_buf(current_data_ptr, range(num_p + 2));
-      buffer temperature_next_buf(next_data_ptr, range(num_p + 2));
-
-      // Iterate over timesteps
-      for (i = 1; i <= num_iter; i++) {
-        if (i % 2 != 0) {
-          q.submit([&](auto& h) {
-            // The size of memory amount that will be given to the buffer.
-            range<1> num_items{num_p + 2};
-
-            accessor temperature(temperature_buf, h);
-            accessor temperature_next(temperature_next_buf, h);
-
-            h.parallel_for(num_items, [=](id<1> k) {
-              size_t gid = k.get(0);
-
-              if (gid == 0) {
-              } else if (gid == num_p + 1) {
-                temperature_next[k] = temperature[k - 1];
-              } else {
-                temperature_next[k] =
-                    C * (temperature[k + 1] - 2 * temperature[k] + temperature[k - 1]) +
-                    temperature[k];
-              }
-            });  // end parallel for loop in kernel1
-          });    // end device queue
-
-        } else {
-          q.submit([&](handler& h) {
-            // The size of memory amount that will be given to the buffer.
-            range<1> num_items{num_p + 2};
-
-            accessor temperature(temperature_buf, h);
-            accessor temperature_next(temperature_next_buf, h);
-
-            h.parallel_for(num_items, [=](id<1> k) {
-              size_t gid = k.get(0);
-
-              if (gid == 0) {
-              } else if (gid == num_p + 1) {
-                temperature[k] = temperature_next[k - 1];
-              } else {
-                temperature[k] = C * (temperature_next[k + 1] - 2 * temperature_next[k] +
-                                  temperature_next[k - 1]) +
-                             temperature_next[k];
-              }
-            });  // end parallel for loop in kernel2
-          });    // end device queue
-        }        // end if %2
-      }          // end iteration
-    }            // end buffer scope
-
-    q.wait_and_throw();
+  ofstream err_file;
+  err_file.open(path);
 
-  } catch (sycl::exception e) {
-    cout << "SYCL exception caught: " << e.what() << "\n";
+  err_file << " \t idx\theat[i]\t\theat_CPU[i] \n";
+
+  for (size_t i = 0; i < num_point + 2; i++) {
+    err_file << "\n RESULT: " << i << "\t" << std::setw(12) << std::left
+             << device_results[i] << "\t" << host_results[i];
```

Comment:
Is there a missing "\n" here?

Reply:
I moved the code, but did not write it. There is a \n at the beginning. I moved it to the end.